domain/application/state/application.go

```diff
@@ -2685,95 +2700,93 @@ ON CONFLICT (application_uuid) DO NOTHING
 		return errors.Errorf("preparing insert application constraints query: %w", err)
 	}
 
-	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		if err := st.checkApplicationNotDead(ctx, tx, appID); err != nil {
-			return errors.Capture(err)
-		}
+	if err := st.checkApplicationNotDead(ctx, tx, appID); err != nil {
+		return errors.Capture(err)
+	}
```

Comment:
We don't need to do this when we insert the application. We should lift this out to only be called when we call `SetApplicationConstraints`.

Reply:
Fixed.